daffodil-core/src/main/scala/org/apache/daffodil/grammar/primitives/SequenceCombinator.scala

```diff
@@ -98,8 +123,27 @@ class UnorderedSequence(sq: SequenceTermBase, sequenceChildrenArg: Seq[SequenceC
 
   import SeparatedSequenceChildBehavior._
 
+  private lazy val sepMtaGram = sq.delimMTA
+  // Note that we actually only every use one of these depending on depending
+  // on various factors. If there is an optional separtor and a suspension is
+  // used to unaprse that separator, then we cannot use the sepMtaUnaprser
+  // because it results in nested suspensions, which isn't allowed. In that
+  // case, the suspension ends up handling both the optional separator and
+  // alignment using sepMtaAlignmentMaybe.
+  private lazy val (sepMtaAlignmentMaybe, sepMtaUnparserMaybe) =
+    if (sepMtaGram.isEmpty) {
+      (MaybeInt.Nope, Maybe.Nope)
+    } else {
+      (MaybeInt(sq.knownEncodingAlignmentInBits), Maybe(sepMtaGram.unparser))
```

Comment:
I think we need a test where the separator requires alignment, but ZL happens and the alignment and separator are both suppressed. A BCD number odd number of nibbles long, separated by DEL (0x7F) characters would do it.

Reply:
Except,... how we get it to be zero length.... That I'm not sure.

Reply:
I've confirmed that we do not have any tests that trigger the situation where we have a optional separator with MTA and a zero length field.

Maybe we can have string fields that are a non-byte size encoding, but separators with byte size encoding? That would require MTA after unparsing a field an before unparsing the separator. And we could have a field that is the empty string, which should be zero length?

Reply:
That is probably the best and easiest way to exercise this functionality.

Reply:
Funny, codecov check seems to say 100% of diff is covered.

Reply:
I think this is a case where code coverage isn't really an accurae representation of test coverage. For example, you could have something like
```scala
if (var1) {
  doTask1()
}

if (var2) {
  doTask2()
}
```
If you have one test where ``var1 == true`` and another test where ``var2 == true``, we'll get 100% line coverage. But we won't get test coverage where both ``var1 == true`` and ``var2 == true``. I think that's what's going on here.

This is partly why we need to rely so heavily on functional tests to ensure correctness--we have so much code that does one individual thing clearly correctly (e.g. a single parser that just parses an int or aligns the bit position) but when combined can lead to incorrect results.

Reply:
I've added a test that ensures that when an optional separator is determined not to be needed due to zero length data, we correctly skip unparsing the mta alignment.